packages/contact-service/src/lib/ContactService.ts

```diff
@@ -0,0 +1,163 @@
+import { EventEmitter } from 'eventemitter3';
+import { ServiceResponse, ServiceResponseStatus } from 'status-service';
+import { Contact } from '../model/Contact';
+import { ContactEventChannel } from '../model/ContactEventChannel';
+import { ContactRepository } from '../repositories/ContactRepository';
+
+/**
+ * ContactService manages contacts and their interactions.
+ * It uses the ContactRepository for data persistence and an EventEmitter
+ * for event-driven communication. The service provides methods to create,
+ * retrieve, update, and delete contacts.
+ */
+export class ContactService {
+  private contactRepository: ContactRepository;
+
+  constructor(private eventBus: EventEmitter) {
+    this.contactRepository = new ContactRepository();
+  }
+
+  /**
+   * Creates a new contact and stores it in the database.
+   * Emits a {@link ContactEventChannel.CreateContact} event upon successful creation.
+   *
+   * @param contact - The contact object to be created (excluding 'id').
+   */
+  public createContact(contact: Omit<Contact, 'id'>): void {
+    const createContactChannel = ContactEventChannel.CreateContact;
+
+    this.contactRepository
+      .create(contact)
+      .then((createdContact) => {
+        const response: ServiceResponse<Contact> = {
+          status: ServiceResponseStatus.Success,
+          payload: createdContact,
+        };
+        this.eventBus.emit(createContactChannel, response);
+      })
+      .catch((error) => {
+        const response: ServiceResponse<Error> = {
+          status: ServiceResponseStatus.Error,
+          payload: error,
+        };
+        this.eventBus.emit(createContactChannel, response);
+      });
```

Comment:
This code snippet is used everywhere. Can you create a private method to create the handler for you?

```typescript
  private sharedErrorHandler(channel: ContactEventChannel) {
    return (error: unknown) => {
      const response: ServiceResponse<Error> = {
        status: ServiceResponseStatus.Error,
        payload: error,
      };

      this.eventBus.emit(channel, response);
    };
  }
  ```